1.2/percentile.go

```diff
@@ -3,25 +3,22 @@ package percentile
 import (
 	"errors"
 	"math"
+	"sort"
 )
 
-type percentile func([]uint) (uint, error)
+type percentile func([]uint) uint
 
-func generatePercentile(x float32) percentile {
+func generatePercentile(x float32) (percentile, error) {
+	if x > 1.0 || x < 0.0 {
+		return nil, errors.New("input out of range")
```

Comment:
In error messages, it is good to show the value that is causing the problem, x in this case.  I'd recommend something like:

`generatePercentile: %q out of range 0 <= x <= 1`

This is building towards the coding challenge which wants "Consistent error handling and error reporting. The system should report clear errors and not crash under non-critical conditions."